src/modules/roc_sndio/target_sox/roc_sndio/sox_source.cpp

```diff
@@ -324,6 +328,11 @@ bool SoxSource::open_() {
         return false;
     }
 
+    if (buffer_size_ == 0) {
```

Comment:
Like in SoxSink, the check is incorrect.

Also, currently SoxSource calls prepare_() and then open_(), so we will use incorrect sample rate here. We should first open device and determine its rate, and only then calculate buffer size and resize buffer.

But SoxSource::prepare_() also contains some code that should be executed *before* open (storing driver and input names). I suggest to split prepare into two functions: setup_names_() and setup_buffer_(). And call them in the following sequence: setup_names_(), open_(), setup_buffer_().

For consistency, let's also rename SoxSink::prepare_() to setup_buffer_().

Reply:
done.